README.md

```diff
@@ -1,112 +1,115 @@
-
-- `3.7`
-  Ice 3.7.x plus various patches (stable, frequently updated)
-
-- `3.6`
-  Ice 3.6.x plus various patches (stable)
-
-- `3.5`
-  Ice 3.5.1 plus various patches (stable)
-
-- `3.4`
-  Ice 3.4.2 plus various patches (stable)
-
-## Copyright and License
-
-Ice is a single-copyright project: all the source code in this [ice
-repository](https://github.com/zeroc-ice/ice) is Copyright &copy; ZeroC, Inc.,
-with very few exceptions.
-
-As copyright owner, ZeroC can license Ice under different license terms, and
-offers the following licenses for Ice:
-- GPL v2, a popular open-source license with strong
-[copyleft](https://en.wikipedia.org/wiki/Copyleft) conditions (the default
-license)
-- Commercial or closed-source licenses
+[![Join the chat at https://gitter.im/zeroc-ice/ice](https://badges.gitter.im/zeroc-ice/ice.svg)][gitter]
+
+# The Ice framework
+
+Ice helps you build networked applications with minimal effort. By taking care of all interactions with low-level
+network programming interfaces, Ice allows you to focus your efforts on your application logic. You don't need to
+worry about details such as opening network connections, encoding and decoding data for network transmission,
+or retrying failed connection attempts (to name just a few of dozens of such low-level details).
+
+[Downloads][downloads] | [Examples][examples] | [Documentation][docs]
+
+## RPCs with Ice
+
+Remote procedure calls ([RPCs][rpcs]) are at the heart of the Ice framework.
+
+You create RPCs with an easy 2-step process:
+1. Define the contract between your client and your server with the [Slice][slice] language—Ice's [IDL][idl].
+2. Run the Slice compiler on these Slice definitions to generate stubs in the programming language(s) of your choice.
+
+For example:
+
+```slice
+// The contract specified using Slice.
+interface Hello
+{
+    // The caller says "hello".
+    void sayHello();
+}
+```
+
+```shell
+# Compile the Slice contract with the Slice compiler for C++ (slice2cpp)
+slice2cpp Hello.ice
+```
+
+```c++
+// C++ client
+
+// Call operation sayHello on a remote object that implements
+// interface Hello using the generated proxy class (HelloPrx).
+helloPrx->sayHello();
+```
+
+```c++
+// C++ server
 
-If you license Ice under GPL v2, there is no license fee or signed license
-agreement: you just need to comply with the GPL v2 terms and conditions. See
-[ICE_LICENSE](./ICE_LICENSE) and [LICENSE](./LICENSE) for further information.
+// Implement the Hello interface by deriving from the generated
+// Hello abstract base class.
+class HelloImpl : public Hello
+{
+public:
+
+    virtual string sayHello(const Ice::Current& current) override
+    {
+        auto ipInfoPtr = dynamic_pointer_cast<Ice::IPConnectionInfo>(
+            current.con->getInfo());
+
+        if (ipInfoPtr)
+        {
+            cout << "Received hello from "
+                 << ipInfoPtr->remoteAddress << ":" << ipInfoPtr->remotePort
+                 << endl;
+        }
```

Comment:
I would just print "Hello world!" instead of getting the IP/port which makes the code more complicated.

Reply:
Yes, I think making it a bit simpler would be good